test/problems/second_quantization/molecular/test_fermionic_op_builder.py

```diff
@@ -0,0 +1,56 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Tests Fermionic Operator builder."""
+from test import QiskitNatureTestCase
+from qiskit_nature.operators import FermionicOp
+from qiskit_nature.problems.second_quantization.molecular import fermionic_op_builder
+from qiskit_nature import QiskitNatureError
+from qiskit_nature.drivers import PySCFDriver, UnitsType
+
+
+# TODO add more detailed tests
+class TestFermionicOperatorBuilder(QiskitNatureTestCase):
+    """Tests Fermionic Operator builder."""
+
+    def setUp(self):
+        super().setUp()
+        try:
+            self.driver = PySCFDriver(atom='Li .0 .0 .0; H .0 .0 1.595',
```

Comment:
We should be able to use a save HDF5 file for this - we have some that already exist. And its easy to create a new one should it be needed. This has the advantage that the tests can be run on any platform and does not require PySCF to be installed which is not possible for instance on Windows. Arguably though here to test the build out - can we not simply use an h1/h2 we define here and ensure the FermionicOp builds out as we expect it to be with the right amount of terms have the expected labels+coeffs. Maybe this is what you have in mind by adding more detailed tests - at which point maybe even the use of an HDF5 driver is not needed unless you want to test with say a larger set of integrals as you would have from LiH.

Reply:
To be honest, I copy-pasted this driver declaration from other tests in Qiskit-nature. I agree that it would be better to make it platform-independent so I might change it to HDF5 as suggested. I just learned about HDF5 yesterday because Max used it in his tests. Perhaps we could open a new issue to address numerous driver inits like this in the repo? Of course if both ways achieve same goals in those contexts.

Reply:
Outside of testing the drivers themselves it would be good to improve matters. Hopefully too we can make these tests much more focused on just testing the modules themselves without too much reliance on other code modules here like drivers.